werror.go

```diff
@@ -232,90 +237,114 @@ func (e *werror) Cause() error {
 	return e.cause
 }
 
+// StackTrace returns the Stacktracer for this error or nil if there is none.
+func (e *werror) StackTrace() StackTrace {
+	return e.stack
+}
+
+// Message returns the message string for this error.
+func (e *werror) Message() string {
+	return e.message
+}
+
 func (e *werror) SafeParams() map[string]interface{} {
-	safe, _ := ParamsFromError(e)
+	safe := make(map[string]interface{})
+	for k, v := range e.params {
```

Comment:
Agreed. Think this is just old code. I wanted to clean it up, but didn't want to lump that into this PR.

Reply:
It looks like the intention is to avoid having the same key in both maps, but I'm not sure whether that's in a spec somewhere or just a decision made a long time ago

Reply:
I realized this is actually incorrect as updated - it's not documented or tested, but the existing behavior is to recursively pull all params from any underlying causes as well, not just this error. Will fix, document, and add tests.